engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -907,6 +912,51 @@ else if (field.getType() == float.class || field.getType() == Float.class ||
         getValue(OperatorContext.METRICS_DIMENSIONS_SCHEME));
     }
 
+    /**
+     * Copy attribute from source attributeMap to destination attributeMap.
+     *
+     * @param dest  destination attribute map.
+     * @param source source attribute map.
+     */
+    private void copyAttributes(AttributeMap dest, AttributeMap source)
+    {
+      for (Entry<Attribute<?>, ?> a : source.entrySet()) {
+        dest.put((Attribute<Object>)a.getKey(), a.getValue());
+      }
+    }
+
+    /**
+     * Copy attribute of operator and port from provided operatorMeta. This function requires
+     * operatorMeta argument is for the same operator.
+     *
+     * @param operatorMeta copy attribute from this OperatorMeta to the object.
+     */
+    private void copyFrom(OperatorMeta operatorMeta)
+    {
+      if (operator != operatorMeta.getOperator()) {
+        throw new IllegalArgumentException("Operator meta is not for the same operator ");
+      }
+
+      // copy operator attributes
+      copyAttributes(attributes, operatorMeta.getAttributes());
+
+      // copy Input port attributes
+      for (Map.Entry<InputPort<?>, InputPortMeta> entry : operatorMeta.getPortMapping().inPortMap.entrySet()) {
+        InputPort<?> key = entry.getKey();
+        InputPortMeta dest = getPortMapping().inPortMap.get(key);
```

Comment:
You don't need key variable. You can directly do `getPortMapping().inPortMap.get(entry.getKey())`